Add lookup_table_v2 BF16 op

wozna opened the pull request:
<!-- Demo: https://github.com/PaddlePaddle/Paddle/pull/24810 -->
### PR types
<!-- One of [ New features | Bug fixes | Function optimization | Performance optimization | Breaking changes | Others ] -->
New features
### PR changes
<!-- One of [ OPs | APIs | Docs | Others ] -->
OPs
### Describe
<!-- Describe what this PR does -->
This PR adds BF16 support for FWD and BWD lookup_table_v2 op. 
Added UT that reuses test `test_lookup_table_bf16_op`.

arogowie-intel commented on python/paddle/fluid/tests/unittests/test_lookup_table_bf16_op.py:
```diff
@@ -25,18 +25,21 @@
 
 
-def _lookup(weights, ids, flat_ids):
+def _lookup(weights, ids, flat_ids, op_type="lookup_table"):
```
Maybe "op_version" would be better? Now it suggests that this function is for different operator types.

jczaja reviewed (approved):
LGTM

lidanqing-vv commented:
@wozna 
```
2021-06-02 16:12:47 2021-06-02 16:12:47 (2.35 MB/s) - 已保存 “bk.txt” [5/5])
2021-06-02 16:12:54 ****************
2021-06-02 16:12:54 0. Developers are not allowed to set the check_dygraph field directly, which is set to True by default. If you need to change the check_dygraph field, you must have one RD (phlrain (Recommend), fuyinno4 (Recommend for kunlun) or lanxianghit) review and approve. 
2021-06-02 16:12:54 The code that do not meet the specification are as follows:
2021-06-02 16:12:54  python/paddle/fluid/tests/unittests/test_lookup_table_v2_bf16_op.py : 
2021-06-02 16:12:54 +        self.check_output_with_place(core.CPUPlace(), check_dygraph=False)
2021-06-02 16:12:54 +        self.check_output_with_place(core.CPUPlace(), check_dygraph=False) 
2021-06-02 16:12:54 1. It is an Op accuracy problem, please take care of it. You must have one RD (zhangting2020 (Recommend), luotao1 or phlrain) approval for the usage (either add or delete) of @skip_check_grad_ci. For more information, please refer to: https://github.com/PaddlePaddle/Paddle/wiki/Gradient-Check-Is-Required-for-Op-Test. The corresponding lines are as follows:
```
Hi, here only approval needed or something else need to be fixed?

wozna commented:
@lidanqing-intel  Yes, only this approval is needed.

lidanqing-vv commented:
@wozna  Please try to set self.is_bfloat16_op() or set op datatype to bf16, to avoid the approval

wozna commented:
> @wozna Please try to set self.is_bfloat16_op() or set op datatype to bf16, to avoid the approval

I removed that, but still, approval for  `skip_check_grad_ci` is needed:
```
@skip_check_grad_ci(
reason="Since paddings are not trainable and fixed in forward,"
"the gradient of paddings makes no sense and we don't "   
 "test the gradient here.")
```

jczaja reviewed (approved):
LGTM

arlesniak reviewed (approved):
LGTM

jczaja commented:
@wzzju Could you please start your review? PR-CI-APPROVAL needs approval to pass.

lidanqing-vv commented:
If approval is too difficult to get, we can only try to use something similar to pass like in test_conv2d_op.py
![image](https://user-images.githubusercontent.com/47151829/121985359-32d5f200-cdc7-11eb-8c6b-bcae1f5fb809.png)

wozna commented:
Initially, I used `@skip_check_grad_ci` exactly as it was in the base tests like in `lookup_table_v2_op.py`. It turned out that even if padding is not used in BWD, checking the gradient in this test works correctly, so I removed that.
